### ISSUE TITLE
Build filter from string

### ISSUE THREAD

**Issue by podhy** [FreeDSx/LDAP#1]
Hello,
it would be nice to be able build filter from string. E.g. I would like to set roles in my application based on users DN and some condition which is saved in my configuration. Something like:

```
$filter = Filters::and(
    Filters::equal('dn, 'my user dn'),
    Filters::raw('memberOf:1.2.840.113556.1.4.1941:=CN=....OU=....DC=...DC=...')
);
```

**Comment by ChadSikorra** [comment]
Yeah, that would be nice. It's something I will add eventually. It requires parsing of the string filter into an actual filter object. As the filter objects are what actually get encoded and sent to the LDAP server for the search. The actual LDAP protocol itself doesn't use these human readable string filter representations. The syntax for it is described in a separate RFC: https://tools.ietf.org/search/rfc4515.

So basically I need to implement a parser based on RFC 4515 to translate these human readable filters into filter objects in this library.

**Comment by ChadSikorra** [comment]
I made more progress on this than expected. This will be the last thing I add before I tag another new version. I just have a tweaks yet and have to go over the code a bit. But the filter string parser is basically complete. It will work just as your example describes with a `Filters::raw($filter)` method.

**Comment by podhy** [comment]
Nice. Thanks great. Btw thanks for this great lib.

**Comment by ChadSikorra** [comment]
Thanks! Added with https://github.com/FreeDSx/LDAP/commit/cfc301117779dea47ae7296fb6b85b0af71c623c. Let me know if you notice any issues if you happen to have the time to try it. I plan on tagging the release later this week.

**Comment by podhy** [comment]
Hello, it works greate. I'm not using advanced filters but I will test it more and if there is problem i give feedback.
